src/H5VLcallback.c

```diff
@@ -1067,13 +1741,28 @@ H5VLattr_create(void *obj, const H5VL_loc_params_t *loc_params, hid_t connector_
     if (NULL == (cls = (H5VL_class_t *)H5I_object_verify(connector_id, H5I_VOL)))
         HGOTO_ERROR(H5E_ARGS, H5E_BADTYPE, NULL, "not a VOL connector ID")
 
-    /* Call the corresponding internal VOL routine */
-    if (NULL == (ret_value = H5VL__attr_create(obj, loc_params, cls, name, type_id, space_id, acpl_id,
-                                               aapl_id, dxpl_id, req)))
-        HGOTO_ERROR(H5E_VOL, H5E_CANTCREATE, NULL, "unable to create attribute")
+    /* Set up context */
+    ctx.obj        = obj;
+    ctx.loc_params = loc_params;
+    ctx.cls        = cls;
+    ctx.name       = name;
+    ctx.type_id    = type_id;
+    ctx.space_id   = space_id;
+    ctx.acpl_id    = acpl_id;
+    ctx.aapl_id    = aapl_id;
+    ctx.dxpl_id    = dxpl_id;
+    ctx.req        = req;
+    ctx.ret_value  = NULL;
+
+    /* Invoke common wrapper routine */
+    if (H5VL__common_api_op(dxpl_id, H5VL__attr_create_api_op, &ctx) < 0)
+        HGOTO_ERROR(H5E_VOL, H5E_CANTOPERATE, NULL, "unable to execute common wrapper operation")
+
```

Comment:
It really adds here to the extra indirection. I don't feel like that's a good direction.